plone/app/z3cform/widgets/text.py

```diff
@@ -0,0 +1,28 @@
+from plone.app.z3cform.interfaces import ITextWidget
+from z3c.form.browser.text import TextWidget as z3cform_TextWidget
+from z3c.form.interfaces import IFieldWidget
+from z3c.form.widget import FieldWidget
+from zope.interface import implementer
+from zope.interface import implementer_only
+
+
+@implementer_only(ITextWidget)
+class TextWidget(z3cform_TextWidget):
+    """ enhanced text widget """
+    klass = "text-widget form-control"
+
+    @property
+    def _html_attributes(self):
+        attributes = super()._html_attributes
+        if self.required:
```

Comment:
To be honest, this was just for testing the extendable attributes. I think it would be better to set the `required` attribute in the template.

Reply:
Ah ok ... the required is missing indeed in the z3c.form FieldProperties. Thanks for the `attributes` insights.